### PR TITLE
Prepare release v0.31.1

### PR DESCRIPTION
**Pull request by erwanor**
Prepare `v0.31.1` to expose `TypedEvent` as intented in #1288. 

* [x] Referenced an issue explaining the need for the change
* [ ] Updated all relevant documentation in docs
* [ ] Updated all code comments where relevant
* [ ] Wrote tests
* [x] Added entry in `.changelog/`

### CODE REVIEW

**Comment by hdevalence** [comment]
I realized I didn't commit the `.changelog` entries from 0.31.0, so I rebased and folded them into that commit, and updated the docs in https://github.com/informalsystems/tendermint-rs/pull/1298/commits/5a6c58c854f5bd86cdef4a60830e204d9c65da53

**Review by hdevalence — APPROVED** [review]

File: .changelog/v0.31.1/summary.md [hunk 1]
```diff
@@ -0,0 +1 @@
+Expose the `TypedEvent` marker trait.
```
Comment (thanethomson) [inline .changelog/v0.31.1/summary.md:1]: We usually have dates associated with releases.

File: .changelog/v0.31.0/summary.md [hunk 2]
```diff
@@ -0,0 +1 @@
+Upgrade signature crate versions and add a `TypedEvent` trait for ABCI events.
```
Comment (thanethomson) [inline .changelog/v0.31.0/summary.md:1]: Also missing a release date here.

File: .changelog/v0.31.1/summary.md (hunk 1 above)
Comment (thanethomson) [reply .changelog/v0.31.1/summary.md:1]: The last time a date was applied was v0.29.0

**Pull request merged**